src/relations/openfga.py

```diff
@@ -264,6 +282,46 @@ def _on_check_auth_rule_action(self, event):
         except ApiException:
             event.set_results({"error": "failed to perform ofga operation"})
 
+    @log_event_handler(logger)
+    def _on_list_system_admins_action(self, event):
+        """Handle OpenFGA list system admins action.
+
+        Args:
+            event: The event triggered when the action is performed.
+        """
+        if not _check_openfga_relation(self.charm._state, event):
+            return
+
+        openfga_data = self.charm._state.openfga
+        admin_groups = self.charm.config["auth-admin-groups"].split(",")
+        results = {key: [] for key in admin_groups}
+
+        if admin_groups == [""]:
+            event.set_results(
+                {"result": "command succeeded", "output": "no admin groups set in 'auth-admin-groups' config"}
+            )
+
+        try:
+            for admin_group in admin_groups:
+                body = ClientExpandRequest(
```

Comment:
Unless groups are allowed to be members of groups, you could also do a more efficient/simple READ requests here to read all the user-* types that are trelated to the `admin_group`

Reply:
Good idea! I've switched this to a READ operation.